### ISSUE TITLE
Investigate why spatial is taking so long to process on Batch

### ISSUE THREAD

**Issue by avrohomgottlieb** [AlexsLemonade/scpca-portal#957]
### Context

In Issue https://github.com/AlexsLemonade/scpca-portal/issues/944, it was discovered that Project SCPCP00006 with the Spatial-SCE configuration ran for a very long time. Even though the job was 10x-20x larger than most of the other project jobs run, this job ran in over 2 hours while the other project jobs ran in around 1-2 minutes.

### Next steps

Investigation should be conducted as to why this job took so long to execute. In order to accomplish this, we should add logging to each function run and try to ascertain where the bottleneck is taking place. Once we know where the bottleneck is, next steps should be determined as to how it should be addressed.

**Comment by avrohomgottlieb** [comment]
### Function run times
| Function | Duration | Notes |
| ---- | ---- | ---- |
| generate_computed_file::generate_computed_file |  02:21:34.162 |
| loader::prep_data_dirs | Less than 00:00:00.000 |
| loader::generate_computed_file | 02:21:34.162 |
| Project::get_computed_file / Project::purge | - | Theses function weren't logged, but likely took around 3 minutes |
| ComputedFile::get_project_file | 02:18:30.864 | | 
| s3::download_input_files | 02:16:51.413 | This function hung for 34 minutes before files began downloading; download speed (of 11.8 GB of files) ranged between 6-9 MiB/s |
| loader::_create_computed_file | 00:03:03.288 |
| s3::upload_output_file |  00:03:03.284 | upload speed of 11.8GB file at ~66 MiB/s |


The log for reference: https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:log-groups/log-group/%2Faws%2Fbatch%2Fjob/log-events/scpca-portal-project-job-definition-avrohom-dev%2Fdefault%2Fac3fea1044c5479f8a7dbbd2bfb49244

**Comment by avrohomgottlieb** [comment]
Regarding the first bug, why `s3::download_input_files` hangs for so long before beginning the downloading, one possible theory is that it's connected to the vast number of individual files that have to be downloaded in one go with project SCPCP000006 compared to other projects. See below how much more numerous project SCPCP000006 is than other projects file-wise.

#### Number of Files per project
| Project  | Number of Files |
| ------- | --------|
| SCPCP000001 | 213 |
| SCPCP000002 | 240 |
| SCPCP000003 | 727 |
| SCPCP000004 | 382 |
| SCPCP000005 | 553 |
| ** SCPCP000006 ** | ** 1766 ** |
| SCPCP000007 | 365 |
| SCPCP000008 | 964 |
| SCPCP000009 | 147 |
| SCPCP000010 | 400 |
| SCPCP000011 | 328 |
| SCPCP000012 | 103 |
| SCPCP000013 | 301 |
| SCPCP000014 | 94 | 
| SCPCP000015 | 121 |
| SCPCP000016 | 67 |
| SCPCP000017 | 249 |
| SCPCP000018 | 103 |
| SCPCP000019 | 0 |
| SCPCP000020 | 36 |
| SCPCP000021 | 139 |
| SCPCP000022 | 454 | 
| SCPCP000023 | 472 |
| SCPCP000024 | 22 |

**Comment by avrohomgottlieb** [comment]
We attempted to configure the aws_cli to improve the downloading throughput with the following config options:

``` 
  max_concurrent_requests = 200
  max_queue_size = 5000
  multipart_threshold = 4MB
  multipart_chunksize = 4MB
```

With this configuration, however, there was no observable difference with the Spatial file.

**Comment by avrohomgottlieb** [comment]
I attempted to change the way that file downloads are conducted with spatial by downloading all files in the `SCPCP000006` directory, instead of passing through 1700 include statements to the `aws s3 sync` command. Instead, I called
```bash 
aws s3 sync scpca-portal-inputs <local_dir> --exclude=* --include=SCPCP000006/*
```

The results were tremendous. Instead of hanging for 34 minutes and then downloading files at 8 MiB/s, the hang time disappeared and throughput jumped up to just below 40 MiB/s. With this adjustment, the run time reduced to 00:17.145. As well, the output file size matched the corresponding file in the staging output bucket.
